### PR TITLE
kiss: remove broken directory symlinks

### LINKED ISSUE #23: Incomplete removal with specific symbolic links
test package:

```
#!/bin/sh -e

install -d "$1/tmp"

cd "$1/tmp"

mkdir 3.20

ln -s 3.20 3.21
```

The broken `3.21`﻿symlink remains when the package is uninstalled due to the logic in `pkg_remove_files`:

```
        # Remove files.
        if [ -f "$file" ] && [ ! -h "$file" ]; then
            rm -f "$file"

        # Remove file symlinks.
        elif [ -h "$file" ] && [ ! -d "$file" ]; then
            rm -f "$file"

        # Remove directories if empty.
        elif [ -d "$file" ] && [ ! -h "$file" ]; then
            rmdir "$file" 2>/dev/null ||:
        fi
```

The condition `[ -h ] && [ ! -d ]` will eval to true only when the `3.21` directory is removed. Here the removal breaks because the check happens before the `3.21` directory is removed. File symlinks aren't affected since they're always deleted.

This can ofcourse be solved by removing the `[ ! -d ]` check, I'm not sure why this check is needed anyways, since directory symlinks can be removed by a simple `rm` too.

### PR DESCRIPTION
**Pull request by dylanaraps**
This changes pkg_remove_files to queue all directory symlinks for
removal *after* all other items. Links are only removed if they are
broken and a safeguard has been added for root level directories.

Also fixed are some bugs around how the paths are stored internally.

1. If KISS_ROOT is empty, path ends up being //path/to/file.
2. Directory symlinks are undetectable when they have '/' as a suffix.

Let me know if there are any issues :)

Closes #23

### CODE REVIEW

File: kiss [hunk 1]
```diff
@@ -1048,21 +1048,28 @@ pkg_remove_files() {
             }
         esac 2>/dev/null ||:
 
-        file=$KISS_ROOT/$file
+        _file=${KISS_ROOT:+"$KISS_ROOT/"}${file%%/}
```
Comment (dilyn-corner) [inline kiss:1051]: Can you explain why this has to happen? Doesn't this just do (for `KISS_ROOT=/test`)
`/test//path/to/file`
?

File: kiss [hunk 2]
```diff
@@ -1048,21 +1048,28 @@ pkg_remove_files() {
...
+        # Remove everything else.
+        else
+            rm -f "$_file"
         fi
     done ||:
+
+    # Remove all broken directory symlinks.
+    for sym do
+        [ -e "$sym" ] || rm -f "$sym"
+    done
```
Comment (dilyn-corner) [inline kiss:1072]: Isn't `unlink` safer? 
`rm -f "$sym"` -> `unlink ${sym%%/}`

Comment (dylanaraps) [reply kiss:1072]: This is broken symlinks only so `rm` is no different no?

File: kiss (hunk 1 above)
Comment (dylanaraps) [reply kiss:1051]: If `KISS_ROOT` is empty this ends up being `//path/to/file`. This only adds the separator when it will be used. In addition, the trailing `/` is stripped from the path as this **breaks** the symlink tests below.

File: kiss (hunk 2 above)
Comment (dilyn-corner) [reply kiss:1072]: I'm asking to cover the corner case I have inevitably failed to identify :)

File: kiss (hunk 1 above)
Comment (dilyn-corner) [reply kiss:1051]: I should have been more clear; is there a reason it has to be //? I know // resolves to / so it doesn't matter in that sense, so I'm confused about why it has to be done -- $file should already have a preceeding /, no?

Comment (dylanaraps) [reply kiss:1051]: I see what you mean. ie, `$KISS_ROOT$file`. Will do some tests locally and fix.

Comment (dylanaraps) [reply kiss]: Fixed (and force pushed).

**Comment by dilyn-corner** [comment]
754c9424f1b74af494b96fcb2dab7d9f51f14557 - Seemed to have no issues :)
